plugins/woocommerce-admin/client/products/hooks/use-product-entity-record.ts

```diff
@@ -15,6 +38,21 @@ export function useProductEntityRecord(
 		undefined
 	);
 
+	useEffect( () => {
+		// This is needed to ensure that we use the correct namespace for the entity data store
+		// without disturbing the rest_namespace outside of the product block editor.
+		wp.apiFetch.use( ( options, next ) => {
+			const versionTwoRegex = new RegExp( '^/wp/v2/product/' );
+			if ( versionTwoRegex.test( options?.path ) && isProductEditor() ) {
+				options.path = options.path.replace(
+					versionTwoRegex,
+					'/wc/v3/products/'
+				);
+			}
+			return next( options );
```

Comment:
This doesn't really need to be in an effect?

Reply:
Not really, although it only needs to run once on the initial render, which this ensures. That said this component likely won't re-render unless the product ID changes and it wouldn't be very consequential if it did.

Reply:
I wonder if it'd be better then to move it outside of the component into the global scope? That would make it slightly more discoverable in the future if it was refactored to be a part of some other initiatlization script. It's not really component level logic.

Reply:
Point taken, although it may be a difference in mental model. My understanding is to keep this as narrow as possible for the time being, targeting only the product block editor, so keeping it close to that logic makes sense in my mind. 

I don't believe we have any other similar logic on the front-end, or it may sense to consolidate that in the same place. That said, I'm receptive to moving this elsewhere.

Reply:
Perhaps a WCA script enqueued on the backend? I'm open to what might feel more appropriate to others.